src/Models/Market.php

```diff
@@ -52,6 +57,16 @@ protected static function boot()
             }
         });
 
+        static::updating(function ($market) {
+            $other_market = Market::where('slug', $market->slug)->where('id', "<>", $market->id)->first();
```

Comment:
This explicit check/custom exception really isn't needed.  The unique constraint on the field will generate an exception already.